code/obj/item/misc_weapons.dm

```diff
@@ -788,6 +788,107 @@
 		. = ..()
 		STOP_TRACKING_CAT(TR_CAT_HUNTER_GEAR)
 
+/obj/item/knife/butterfly
+	name = "butterfly knife"
+	desc = "A small, compact butterfly knife. Concealable when folded."
+	icon = 'icons/obj/items/weapons.dmi'
+	inhand_image_icon = 'icons/mob/inhand/hand_weapons.dmi'
+	icon_state = "butterfly-knife"
+	uses_multiple_icon_states = 1
+	var/active = 0 // 0 = closed, 1 = open
+	hit_type = DAMAGE_BLUNT
+	flags = ATTACK_SELF_DELAY | CONDUCT | TABLEPASS
+	w_class = W_CLASS_SMALL
+	force = 4
+	throwforce = 10
+	throw_speed = 4
+	throw_range = 8
+	click_delay = 0.7 SECONDS
+
+	New()
+		..()
+		RegisterSignal(src, COMSIG_ITEM_TWIRLED, .proc/sicktricks)
+
+/obj/item/knife/butterfly/proc/sicktricks()
+	if (src.active)
+		flick("butterfly-knife_trick", src)
+	else
+		flick("butterfly-knife_trickclosed", src)
+
+
+
+
+/obj/item/knife/butterfly/attack_self(mob/user as mob)
+	if (src.active)
+		src.active = !( src.active )
```

Comment:
```suggestion
		src.active = !src.active
```
you don't need the brackets, also consider moving the toggle outside the if statement as it happens in both if and else cases